components/forms/src/api.rs

```diff
@@ -0,0 +1,553 @@
+                email: row.get("email")?,
+            };
+            Ok(Address {
+                guid: Guid::from_string(row.get("guid")?),
+                address_fields: new_address,
+                time_created: row.get("time_created")?,
+                time_last_used: row.get("time_last_used")?,
+                time_last_modified: row.get("time_last_modified")?,
+                times_used: row.get("times_used")?,
+                sync_change_counter: row.get("sync_change_counter")?,
+            })
+        })?;
+
+        for address_result in addresses_iter {
+            addresses.push(address_result.expect("Should unwrap address"));
+        }
+    }
+
+    tx.commit()?;
+    Ok(addresses)
+}
+
+#[allow(dead_code)]
+pub fn update_address(conn: &mut Connection, address: Address) -> Result<()> {
+    let tx = conn.transaction()?;
+    tx.execute_named(
+        "UPDATE addresses_data
+         SET given_name     = :given_name,
+            additional_name = :additional_name,
+            family_name     = :family_name,
+            organization    = :organization,
+            street_address  = :street_address,
+            address_level3  = :address_level3,
+            address_level2  = :address_level2,
+            address_level1  = :address_level1,
+            postal_code     = :postal_code,
+            country         = :country,
+            tel             = :tel,
+            email           = :email
+        WHERE guid          = :guid",
+        rusqlite::named_params! {
+            ":given_name": address.address_fields.given_name,
+            ":additional_name": address.address_fields.additional_name,
+            ":family_name": address.address_fields.family_name,
+            ":organization": address.address_fields.organization,
+            ":street_address": address.address_fields.street_address,
+            ":address_level3": address.address_fields.address_level3,
+            ":address_level2": address.address_fields.address_level2,
+            ":address_level1": address.address_fields.address_level1,
+            ":postal_code": address.address_fields.postal_code,
+            ":country": address.address_fields.country,
+            ":tel": address.address_fields.tel,
+            ":email": address.address_fields.email,
+            ":guid": address.guid,
+        },
+    )?;
+
+    tx.commit()?;
+    Ok(())
+}
+
+#[allow(dead_code)]
+fn delete_address(conn: &mut Connection, guid: &Guid) -> Result<bool> {
+    let tx = conn.transaction()?;
+
+    // check that guid exists
+    let exists = tx.query_row(
+        "SELECT EXISTS (
+            SELECT 1
+            FROM addresses_data d
+            WHERE guid = :guid
+                AND NOT EXISTS (
+                    SELECT 1
+                    FROM   addresses_tombstones t
+                    WHERE  d.guid = t.guid
+                )
+        )",
+        &[guid.as_str()],
+        |row| row.get(0),
+    )?;
+
+    if exists {
```

Comment:
Hmmm, I think this is missing a `DELETE FROM addresses_data` to drop the row?

Reply:
I honestly wasn't sure how this worked so thank you for the clarification. And for sure, as you stated below, this simplifies the "exists" check.